core/list.go

```diff
@@ -255,6 +255,8 @@ func (lb *ListBase) Init() {
 		// absorb horizontal here, vertical in view
 		s.Overflow.X = styles.OverflowAuto
 		s.Grow.Set(1, 1)
+		svi := lb.This.(Lister)
+		svi.UpdateMaxWidths()
```

Comment:
Can you give a coherent explanation of why this needs to go in `Styler` and not `Updater`?

Reply:
styler is only run when there are more significant changes; Update is run to redraw during scrolling given the nature of the way this widget works, with static widgets representing different content as it scrolls.

Reply:
That is a complete inversion of normal expectations; it is the case for every other widget that `Update` is a much more major process and `Style` is run more frequently. Are you re-running the `Maker` while you scroll? I thought the whole point was to reuse the same widgets for the table. It seems like there should be a better mechanism than `UpdateWidget` for updating while scrolling.

Reply:
I can never remember what happens when.. Update does run Style, and in fact the slow UpdateMaxWidths call is being made in the Style. I'm looking into it.

Reply:
ok now calling UpdateWidths precisely when needed.  tested to 10_000_000 rows with no problems!